src/test/java/org/truffleruby/tck/RubyDebugTest.java

```diff
@@ -177,6 +180,37 @@ public void stepInStepOver() throws Throwable {
         assertExecutedOK("Stepping went OK");
     }
 
+    @Test
+    public void testReenterArgumentsAndValues() throws Throwable {
+        // Test that after a re-enter, arguments are kept and variables are cleared.
+        final Source source = createReenterValuesTest();
+
+        run.addLast(() -> {
+            assertNull(suspendedEvent);
+            assertNotNull(debuggerSession);
+            debuggerSession.install(Breakpoint.newBuilder(source).lineIs(12).build());
+        });
+        engine.eval(source);
+
+        // TODO: assertArguments(12, "x = n + m", "n", "11", "m", "20");
```

Comment:
I like our attribution system for `TODO` comments.

Reply:
Fixing.